### ISSUE TITLE
TypeScript new line default on Windows breaks on *nix

### ISSUE THREAD

**Issue by leidegre** [Microsoft/TypeScript#29928]
I would like to purpose that this is a bug because it causes needless trouble in cross platform environments and I cannot find a single reason for why we insist on having `CR` `LF` on Windows any more.

**Expected behavior:**
Linux/OS X users don't run into subtle issues because of carriage return in bin scripts.

**Actual behavior:**
When a TypeScript project is built on Windows and then published to NPM, the default new line options is `CR` `LF`. If your `package.json` has a `bin` script with a shebang like this `#!/usr/bin/env node` TypeScript will rewrite this with a `CR` `LF` line ending which breaks on at least Mac OS X with the following error `env: node\r: No such file or directory
error Command failed with exit code 127` (haven't tested this elsewhere).

I'm raising this as a bug because it has non obvious implications for people developing software with TypeScript if they publish their code to NPM. And I don't think that there is a valid reason for why `CR` `LF` should be used on Windows at all. It's an archaic historical artifact that should have been laid to rest a very long time ago.

I'm going to be bold and suggest that the option `--newLine` should never have been added. That it should be deprecated and that the option should be ignored. And that TypeScript only emit `LF` from now on.

**Comment by j-oliveras** [comment]
This options already exists (from [compiler options](https://www.typescriptlang.org/docs/handbook/compiler-options.html)):

| Option | Type | Default | Description |
| -- | -- | -- | -- |
| --newLine | string | (platform specific) | Use the specified end of line sequence to be used when emitting files: "crlf" (windows) or "lf" (unix).” |

**Comment by leidegre** [comment]
@j-oliveras please read the last paragraph.

**Comment by RyanCavanaugh** [comment]
There are certainly other people which have workflows which depend on CR+LF. Configuration is hard; making it harder on purpose by removing options or breaking backward compatibility is not a priority for us.

**Comment by leidegre** [comment]
@RyanCavanaugh I respect your decision. I disagree wholeheartedly.

Ultimately, I think this leaves TypeScript in a worse place because it becomes just another thing you list of exceptions to consider, albeit only if you use Windows as a web development platform. The use case that you have effectively broken with this design decision is writing TypeScript on Windows and then publishing to NPM. It won't work without the `--newLine` option set to `LF`.

I don't think it should work like that. But what do I know, I'm just a guy with an opinion.

I would like to end this by just saying that I really enjoy TypeScript. And I appreciate all the good work that has went into TypeScript. It's made me look at typing stuff in a new way. And I'm not trying to be salty, I genuinely believe that what I try to argue here would make TypeScript better.

**Comment by leidegre** [comment]
> There are certainly other people which have workflows which depend on CR+LF. Configuration is hard; making it harder on purpose by removing options or breaking backward compatibility is not a priority for us.

@RyanCavanaugh you give me one example and I will believe it. 😄

**Comment by gmeligio** [comment]
@leidegre In case it helps you, the new default in TS 5.5 is `LF`, according to the release notes. It might improve your use cases.
https://github.com/microsoft/TypeScript/issues/51909
https://www.typescriptlang.org/docs/handbook/compiler-options.html

I just found this issue after reading the release notes of 5.5. I noticed this `newline` option that I wasn't aware of and started investigating its usage.
